test/main.tf

```diff
@@ -0,0 +1,67 @@
+# Basic terraform tests. If they break, you probably have introduced a breaking change.
+# These tests cannot catch errors, but can be used to verify that the more complex variables worked as expected.
+
+provider "aws" {
+  region  = var.region
+  profile = var.profile
+}
+
+# Defaults, except no multi-region cloudtrail to avoid hitting 5 trail limit
+module "test_module_1" {
+  source           = "../"
+  name             = "${var.name}-1"
+  observe_customer = "123"
+  observe_token    = "ds123456789:dogdogdogdog"
+
+  cloudtrail_is_multi_region_trail = false
+}
+
+locals {
+  result1 = {
+    "cloudtrail" : length(module.test_module_1.cloudtrail) > 0,
```

Comment:
What is this trying to do? Checking for the `length` of a resource is unexpected and its behavior shouldn't be relied on. `terraform apply` would error if the trail resource could not be created, so when does this condition ever not pass?

Reply:
the cloudtrail resource involves `count` so there is either 1 or 0 elements

Reply:
Ah right that makes a lot more sense. `length` with `for_each` or `count` is good. Question still stands re: `terraform apply` though. It seems like the goal is to test that the module was applied, not the conditional logic of trail creation.

Reply:
this test checks that the module input variables work as intended. We would expect a CloudTrail to be created by the module.